index.ts

```diff
@@ -68,14 +69,30 @@ export class HttpClientResponse implements ifm.IHttpClientResponse {
     public message: http.IncomingMessage;
     readBody(): Promise<string> {
```

Comment:
I'm still trying to figure out why `message` can't reliably be used to read a gzipped response. But given what you have I think it would be good to add a public `readBodyAsBuffer` to make it easier to write the response body, regardless of encoding, to a stream / file. `readBody()` could call `readBodyAsBuffer` and convert the buffer to a string.